### PR TITLE
Adds call to $file->close() in Shell::createFile()

### PR DESCRIPTION
**Pull request by thinkingmedia**
I found that PHP in the CLI was leaving the file handle open for `createFile`. For most people, this isn't an issue as it will close when PHP exists, but I was batch processing a few thousand files when I ran into this as a memory leak.

Closing the file resolved the leak.

### CODE REVIEW

**Comment by thinkingmedia** [comment]
Is there a way I can squash the above commits into one without having to restart the PR?

File: src/Console/Shell.php [hunk 1]
```diff
@@ -806,14 +806,20 @@ public function createFile($path, $contents)
...
         $File = new File($path, true);
-        if ($File->exists() && $File->writable()) {
-            $File->write($contents);
-            $this->_io->out(sprintf('<success>Wrote</success> `%s`', $path));
-            return true;
-        }
 
-        $this->_io->err(sprintf('<error>Could not write to `%s`</error>.', $path), 2);
-        return false;
+        try
```
Comment (thinkingmedia) [inline src/Console/Shell.php]: I hate you. lol

**Comment by markstory** [comment]
@thinkingmedia Yes.

```bash
git checkout master
git pull upstream master
git checkout thinkingmedia/close_file
git rebase -i master
git push --force origin
```

**Comment by thinkingmedia** [comment]
okay, I'm done.

**Comment by dereuromark** [comment]
Nice finding :+1:

**Comment by chinpei215** [comment]
Weird. `$File` seems to be a local variable. Why does memory leak occur, in spite of the fact that the `close()` method is called in the [__destruct()](https://github.com/cakephp/cakephp/blob/4a9249fe23be41850d5b0e23f19dce729e63f1d8/src/Filesystem/File.php#L97-L100)?

**Comment by thinkingmedia** [comment]
@chinpei215 that's what I thought as well, but what I found is that it wouldn't close the file until garbage collection is done.

**Pull request merged**